Building a Reqnroll project on macOS ARM64 architecture (eg MacBook M1) fails

### Reqnroll Version

2.0.0

### Which test runner are you using?

MSTest

### Test Runner Version Number

n/a

### .NET Implementation

.NET 8.0

### Test Execution Method

Other – PLEASE SPECIFY

### Content of reqnroll.json configuration file

_No response_

### Issue Description

It seems that Reqnroll v2.0.0 projects fail to build on MacOS ARM64 processor machines. 

The error message is similar to this:
```
/Users/<user>/.nuget/packages/reqnroll.mstest/2.0.0/build/Reqnroll.MsTest.targets(34,5): error MSB4216: Could not run the "ReplaceTokenInFileTask" task because MSBuild could not create or connect to a task host with runtime "NET" and architecture "arm64".  Please ensure that (1) the requested runtime and/or architecture are available on the machine, and (2) that the required executable "MSBuild.exe" exists and can be run. [/Users/seb/Projects/Quickstart/ReqnrollQuickstart.Specs/ReqnrollQuickstart.Specs.csproj]
    0 Warning(s)
    1 Error(s)
```

Related discussion: https://github.com/orgs/reqnroll/discussions/143

### Steps to Reproduce

* Get a MacBook M1
* Clone Reqnroll Quickstart (https://github.com/reqnroll/Quickstart)
* Run: `dotnet build`


### Link to Repro Project

_No response_

--- a/.gitignore
+++ b/.gitignore
@@ -367,8 +367,6 @@ Reqnroll.sln.GhostDoc.xml
 Features.Generated/
 GitExtensions.settings.backup
 
-/Installer/NuGetPackages/Reqnroll.Tools.MsBuild.Generation/build/Reqnroll.Tools.MsBuild.Generation.targets
-/Installer/NuGetPackages/Reqnroll.Tools.MsBuild.Generation/build/Reqnroll.Tools.MsBuild.Generation.props
 /Tests/Reqnroll.MsBuildNetSdk.IntegrationTests/Features/dummy.feature.cs
 *.feature.cs
 *.AssemblyHooks.cs

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # [vNext]
 
+## Bug fixes:
+
+* Fix: Building a Reqnroll project on macOS ARM64 architecture (eg MacBook M1) fails (#152)
+
 # v2.0.1 - 2024-05-29
 
 ## Bug fixes:

--- a/Plugins/Reqnroll.ExternalData/Reqnroll.ExternalData.ReqnrollPlugin.IntegrationTest/Reqnroll.ExternalData.ReqnrollPlugin.IntegrationTest.csproj
+++ b/Plugins/Reqnroll.ExternalData/Reqnroll.ExternalData.ReqnrollPlugin.IntegrationTest/Reqnroll.ExternalData.ReqnrollPlugin.IntegrationTest.csproj
@@ -1,7 +1,7 @@
 ﻿<Project>
 
   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
-  <Import Project="..\..\..\Reqnroll.Tools.MsBuild.Generation\build\Reqnroll.Tools.MsBuild.Generation.props" />
+  <Import Project="..\..\..\Tests\Reqnroll.Specs\build/Reqnroll.Tools.MsBuild.Generation.OwnTests.props" />
 
   <PropertyGroup>
     <TargetFramework>net8.0</TargetFramework>
@@ -23,11 +23,6 @@
 
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
 
-  <PropertyGroup>
-    <_Reqnroll_Needed_MSBuildGenerator Condition=" '$(MSBuildRuntimeType)' == 'Core'">$(Reqnroll_Core_Tools_TFM)</_Reqnroll_Needed_MSBuildGenerator>
-    <_Reqnroll_Needed_MSBuildGenerator Condition=" '$(MSBuildRuntimeType)' != 'Core'">$(Reqnroll_FullFramework_Tools_TFM)</_Reqnroll_Needed_MSBuildGenerator>
-  </PropertyGroup>
-
   <Target Name="PreBuild">
     <MSBuild Projects="..\..\..\Reqnroll.Tools.MsBuild.Generation\Reqnroll.Tools.MsBuild.Generation.csproj" ContinueOnError="false" Properties="TargetFramework=$(_Reqnroll_Needed_MSBuildGenerator)" />
     <MSBuild Projects="..\Reqnroll.ExternalData.ReqnrollPlugin\Reqnroll.ExternalData.ReqnrollPlugin.csproj" ContinueOnError="false" Properties="TargetFramework=netstandard2.0" />
@@ -36,10 +31,6 @@
   <ItemGroup>
     <ReqnrollGeneratorPlugins Include="$(MSBuildThisFileDirectory)..\Reqnroll.ExternalData.ReqnrollPlugin\bin\$(Configuration)\netstandard2.0\Reqnroll.ExternalData.ReqnrollPlugin.dll" />
   </ItemGroup>
-
-  <PropertyGroup>
-    <_Reqnroll_TaskAssembly>..\..\Reqnroll.Tools.MsBuild.Generation\bin\$(Configuration)\$(_Reqnroll_Needed_MSBuildGenerator)\tasks\Reqnroll.Tools.MsBuild.Generation.dll</_Reqnroll_TaskAssembly>
-  </PropertyGroup>
 
   <Import Project="..\..\..\Reqnroll.Tools.MsBuild.Generation\build\Reqnroll.Tools.MsBuild.Generation.targets" />
   <!-- has to be before the PropertyGroup change-->

--- a/Plugins/Reqnroll.Verify/Reqnroll.Verify.ReqnrollPlugin.IntegrationTest/Reqnroll.Verify.ReqnrollPlugin.IntegrationTest.csproj
+++ b/Plugins/Reqnroll.Verify/Reqnroll.Verify.ReqnrollPlugin.IntegrationTest/Reqnroll.Verify.ReqnrollPlugin.IntegrationTest.csproj
@@ -1,7 +1,7 @@
 ﻿<Project>
 
   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
-  <Import Project="..\..\..\Reqnroll.Tools.MsBuild.Generation\build\Reqnroll.Tools.MsBuild.Generation.props" />
+  <Import Project="..\..\..\Tests\Reqnroll.Specs\build/Reqnroll.Tools.MsBuild.Generation.OwnTests.props" />
 
   <PropertyGroup>
     <TargetFramework>net8.0</TargetFramework>
@@ -30,9 +30,6 @@
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
 
   <PropertyGroup>
-    <_Reqnroll_Needed_MSBuildGenerator Condition=" '$(MSBuildRuntimeType)' == 'Core'">$(Reqnroll_Core_Tools_TFM)</_Reqnroll_Needed_MSBuildGenerator>
-    <_Reqnroll_Needed_MSBuildGenerator Condition=" '$(MSBuildRuntimeType)' != 'Core'">$(Reqnroll_FullFramework_Tools_TFM)</_Reqnroll_Needed_MSBuildGenerator>
-
     <_VerifyGeneratorPluginFramework Condition=" '$(MSBuildRuntimeType)' == 'Core'">netstandard2.0</_VerifyGeneratorPluginFramework>
     <_VerifyGeneratorPluginFramework Condition=" '$(MSBuildRuntimeType)' != 'Core'">net462</_VerifyGeneratorPluginFramework>
   </PropertyGroup>
@@ -51,10 +48,6 @@
   <ItemGroup>
     <Folder Include="Features\" />
   </ItemGroup>
-
-  <PropertyGroup>
-    <_Reqnroll_TaskAssembly>..\..\Reqnroll.Tools.MsBuild.Generation\bin\$(Configuration)\$(_Reqnroll_Needed_MSBuildGenerator)\tasks\Reqnroll.Tools.MsBuild.Generation.dll</_Reqnroll_TaskAssembly>
-  </PropertyGroup>
 
   <Import Project="..\..\..\Reqnroll.Tools.MsBuild.Generation\build\Reqnroll.Tools.MsBuild.Generation.targets" />
   <!-- has to be before the PropertyGroup change-->

--- a/Reqnroll.Tools.MsBuild.Generation/build/Reqnroll.Tools.MsBuild.Generation.tasks
+++ b/Reqnroll.Tools.MsBuild.Generation/build/Reqnroll.Tools.MsBuild.Generation.tasks
@@ -1,13 +1,6 @@
 <Project>
-  <!-- Using `TaskHostFactory` ensures that the task assembly will not be locked
-       at the end of the build. If it's omitted, that can cause build failures
-       in subsequent builds because the task assembly can't be written by the
-       next build.
-
-       Reference: https://learn.microsoft.com/en-us/visualstudio/msbuild/how-to-configure-targets-and-tasks?view=vs-2022#task-factories
-  -->
-  <UsingTask TaskName="Reqnroll.Tools.MsBuild.Generation.GenerateFeatureFileCodeBehindTask" AssemblyFile="$(_Reqnroll_TaskAssembly)" TaskFactory="TaskHostFactory" />
-  <UsingTask TaskName="Reqnroll.Tools.MsBuild.Generation.ReplaceTokenInFileTask" AssemblyFile="$(_Reqnroll_TaskAssembly)" TaskFactory="TaskHostFactory" />
+  <UsingTask TaskName="Reqnroll.Tools.MsBuild.Generation.GenerateFeatureFileCodeBehindTask" AssemblyFile="$(_Reqnroll_TaskAssembly)" />
+  <UsingTask TaskName="Reqnroll.Tools.MsBuild.Generation.ReplaceTokenInFileTask" AssemblyFile="$(_Reqnroll_TaskAssembly)" />
 
   <PropertyGroup>
     <_ReqnrollTasksImported>true</_ReqnrollTasksImported>

--- a/Tests/Reqnroll.Specs/Reqnroll.Specs.csproj
+++ b/Tests/Reqnroll.Specs/Reqnroll.Specs.csproj
@@ -1,7 +1,7 @@
 ﻿<Project>
 
   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
-  <Import Project="../../Reqnroll.Tools.MsBuild.Generation/build/Reqnroll.Tools.MsBuild.Generation.props" />
+  <Import Project="build/Reqnroll.Tools.MsBuild.Generation.OwnTests.props" />
 
   <PropertyGroup>
     <TargetFramework>net8.0</TargetFramework>
@@ -56,13 +56,6 @@
 
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
 
-  <PropertyGroup>
-    <_Reqnroll_Needed_MSBuildGenerator Condition=" '$(MSBuildRuntimeType)' == 'Core'">$(Reqnroll_Core_Tools_TFM)</_Reqnroll_Needed_MSBuildGenerator>
-    <_Reqnroll_Needed_MSBuildGenerator Condition=" '$(MSBuildRuntimeType)' != 'Core'">$(Reqnroll_FullFramework_Tools_TFM)</_Reqnroll_Needed_MSBuildGenerator>
-  </PropertyGroup>
-
-
-
   <Target Name="WriteNuGetVersion" AfterTargets="GetBuildVersion" BeforeTargets="BeforeCompile" Condition="$(DesignTimeBuild) != 'true' OR '$(BuildingProject)' == 'true'">
     <ReplaceTextInFileTask InputFile="$(ProjectDir)\\NuGetPackageVersion.template.cs" OutputFile="$(ProjectDir)\\NuGetPackageVersion.cs" TextToReplace="NUGET_VERSION" TextToReplaceWith="$(PackageVersion)" WriteOnlyWhenChanged="true" />
 
@@ -79,10 +72,6 @@
   <ItemGroup>
     <ReqnrollGeneratorPlugins Include="$(MSBuildThisFileDirectory)..\..\Plugins\Reqnroll.xUnit.Generator.ReqnrollPlugin\bin\$(Configuration)\netstandard2.0\Reqnroll.xUnit.Generator.ReqnrollPlugin.dll" />
   </ItemGroup>
-
-  <PropertyGroup>
-    <_Reqnroll_TaskAssembly>..\..\Reqnroll.Tools.MsBuild.Generation\bin\$(Configuration)\$(_Reqnroll_Needed_MSBuildGenerator)\tasks\Reqnroll.Tools.MsBuild.Generation.dll</_Reqnroll_TaskAssembly>
-  </PropertyGroup>
 
   <Import Project="..\..\Reqnroll.Tools.MsBuild.Generation\build\Reqnroll.Tools.MsBuild.Generation.targets" /> <!-- has to be before the PropertyGroup change-->
   <PropertyGroup>

--- a/Tests/Reqnroll.Specs/build/Reqnroll.Tools.MsBuild.Generation.OwnTests.props
+++ b/Tests/Reqnroll.Specs/build/Reqnroll.Tools.MsBuild.Generation.OwnTests.props
@@ -1,0 +1,18 @@
+<Project TreatAsLocalProperty="TaskFolder;TaskAssembly">
+
+  <PropertyGroup>
+    <_ReqnrollTasksImported>true</_ReqnrollTasksImported>
+  </PropertyGroup>
+
+  <Import Project="../../../Reqnroll.Tools.MsBuild.Generation\build\Reqnroll.Tools.MsBuild.Generation.props" />
+
+  <PropertyGroup>
+    <_Reqnroll_Needed_MSBuildGenerator Condition=" '$(MSBuildRuntimeType)' == 'Core'">$(Reqnroll_Core_Tools_TFM)</_Reqnroll_Needed_MSBuildGenerator>
+    <_Reqnroll_Needed_MSBuildGenerator Condition=" '$(MSBuildRuntimeType)' != 'Core'">$(Reqnroll_FullFramework_Tools_TFM)</_Reqnroll_Needed_MSBuildGenerator>
+  </PropertyGroup>
+  <PropertyGroup>
+    <_Reqnroll_TaskAssembly>$(MSBuildThisFileDirectory)../../../Reqnroll.Tools.MsBuild.Generation/bin/$(Configuration)/$(_Reqnroll_Needed_MSBuildGenerator)/tasks/Reqnroll.Tools.MsBuild.Generation.dll</_Reqnroll_TaskAssembly>
+  </PropertyGroup>
+  <Import Project="Reqnroll.Tools.MsBuild.Generation.OwnTests.tasks" Condition="'$(_ReqnrollTasksOwnTestImported)' =='' " />
+
+</Project>

--- a/Tests/Reqnroll.Specs/build/Reqnroll.Tools.MsBuild.Generation.OwnTests.tasks
+++ b/Tests/Reqnroll.Specs/build/Reqnroll.Tools.MsBuild.Generation.OwnTests.tasks
@@ -1,0 +1,16 @@
+<Project>
+  <!-- Using `TaskHostFactory` ensures that the task assembly will not be locked
+       at the end of the build. If it's omitted, that can cause build failures
+       in subsequent builds because the task assembly can't be written by the
+       next build.
+
+       Reference: https://learn.microsoft.com/en-us/visualstudio/msbuild/how-to-configure-targets-and-tasks?view=vs-2022#task-factories
+  -->
+  <UsingTask TaskName="Reqnroll.Tools.MsBuild.Generation.GenerateFeatureFileCodeBehindTask" AssemblyFile="$(_Reqnroll_TaskAssembly)" TaskFactory="TaskHostFactory" />
+  <UsingTask TaskName="Reqnroll.Tools.MsBuild.Generation.ReplaceTokenInFileTask" AssemblyFile="$(_Reqnroll_TaskAssembly)" TaskFactory="TaskHostFactory" />
+
+  <PropertyGroup>
+    <_ReqnrollTasksImported>true</_ReqnrollTasksImported>
+    <_ReqnrollTasksOwnTestImported>true</_ReqnrollTasksOwnTestImported>
+  </PropertyGroup>
+</Project>

